daemon/algod/api/server/v2/handlers.go

```diff
@@ -1152,3 +1152,27 @@ func (v2 *Handlers) TealCompile(ctx echo.Context) error {
 	}
 	return ctx.JSON(http.StatusOK, response)
 }
+
+// TealDisassemble disassembles the program bytecode in base64 into TEAL code.
+// (POST /v2/teal/disassemble)
+func (v2 *Handlers) TealDisassemble(ctx echo.Context) error {
+	// return early if teal compile is not allowed in node config
+	if !v2.Node.Config().EnableDeveloperAPI {
+		return ctx.String(http.StatusNotFound, "/teal/disassemble was not enabled in the configuration file by setting the EnableDeveloperAPI to true")
+	}
+	buf := new(bytes.Buffer)
+	ctx.Request().Body = http.MaxBytesReader(nil, ctx.Request().Body, maxTealSourceBytes)
+	buf.ReadFrom(ctx.Request().Body)
```

Comment:
@algochoi I opened an optional PR for your consideration:  https://github.com/algochoi/go-algorand/pull/3.  

I don't know enough about Go specifics to feel confident the proposed change is appropriate.  I'd appreciate your review to help make a call.

If the change isn't worthwhile, I'll happily close the thread + child PR.

Reply:
@algochoi Thanks for brining in the change - resolving.